synapse/logging/opentracing.py

```diff
@@ -733,37 +733,59 @@ def decorator(func):
 
         _opname = opname if opname else func.__name__
 
-        @wraps(func)
-        def _trace_inner(*args, **kwargs):
-            if opentracing is None:
-                return func(*args, **kwargs)
+        if inspect.iscoroutinefunction(func):
+            @wraps(func)
+            async def _trace_inner(*args, **kwargs):
+                if opentracing is None:
+                    return await func(*args, **kwargs)
```

Comment:
This is done below also (in the other `_trace_inner` function), but we check this at the top of the decorator, above so I think it can be removed from both?

Reply:
yes

Reply:
Done! Thanks. 👍

Reply:
Turns out these are necessary since opentracing can be installed, but then disabled.

https://github.com/matrix-org/synapse/blob/f58a80e8a410d0b19d5c3a79c491e8d3248b89c0/synapse/logging/opentracing.py#L331-L335